src/app/(main)/connections/_components/modal.tsx

```diff
@@ -13,16 +11,14 @@ import {
 import { Input } from "@scylla-studio/components/ui/input";
 import { Modal } from "@scylla-studio/components/ui/modal";
 import { Plus } from "lucide-react";
-import { ReactNode, useState, useTransition } from "react";
+import React from "react";
+import { ReactNode, useEffect, useState, useTransition } from "react";
 import { FormProvider, useForm } from "react-hook-form";
 import { z } from "zod";
 
 const formSchema = z.object({
   name: z.string().min(1, { message: "Name is required." }),
-  host: z
-    .string()
-    .min(1, { message: "Host is required." })
-    .regex(/^[a-zA-Z0-9.-]+$/, "Invalid host format."),
+  host: z.string().ip(),
   username: z.string().min(1, { message: "Username is required." }),
   password: z.string().min(1, { message: "Password is required." }),
   dc: z.string().min(1, { message: "Data center (DC) is required." }),
```

Comment:
I forgot to mention, but not all items on that table is actually a "fillable". @Daniel-Boll please correct me, but we can retrieve the DC structure directly from the Driver. Correct? This will enable us to know by a single connection the topology of the entire cluster after the first connection.

Reply:
Ngl, I don't remember, never used the DC structure query, but we may be able to get it from the driver.

Reply:
We can check it tomorrow then.